server/enterprise/message_export/worker.go

```diff
@@ -362,6 +366,35 @@ func (w *MessageExportWorker) initJobData(logger mlog.LoggerIFace, job *model.Jo
 	job.Data[shared.JobDataExportDir] = getJobExportDir(logger, job.Data, job.Data[shared.JobDataJobStartTime], job.Data[shared.JobDataJobEndTime])
 }
 
+func (w *MessageExportWorker) getPreviousJob(rctx request.CTX) (*model.Job, error) {
+	offset := 0
+
+	for {
+		jobs, err := w.jobServer.Store.Job().GetAllByTypesAndStatusesPage(rctx,
+			[]string{model.JobTypeMessageExport},
+			[]string{model.JobStatusWarning, model.JobStatusSuccess},
+			offset, DefaultPreviousJobPageSize)
+		if err != nil {
+			return nil, err
+		}
+
+		// If no more jobs are found, we've reached the end
+		if len(jobs) == 0 {
```

Comment:
if len(jobs) < DefaultPreviousJobPageSize, that is also the end right?

Reply:
Six of one, half dozen of the other. But your way saves a db call, so that wins. :)